src/Shared/HttpExtensions.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+
+internal static class HttpExtensions
+{
+    internal static bool IsValidHttpMethodForForm(string method) =>
+        HttpMethods.IsPost(method) || HttpMethods.IsPut(method) || HttpMethods.IsPatch(method);
+
+    internal static bool IsValidContentTypeForForm(string? contentType) =>
+        MediaTypeHeaderValue.TryParse(contentType, out var mediaType) &&
```

Comment:
This allocates, no?

Reply:
Mmmm, yeah, and we would be paying the cost here for most requests (`POST`, `PUT`). I think I can cook up something better here.